be/src/runtime/mem_tracker.h

```diff
@@ -298,6 +298,8 @@ class MemTracker {
 
     bool limit_exceeded_by_ratio(int64_t ratio) const { return _limit >= 0 && (_limit * ratio / 100) < consumption(); }
 
+    bool limit_exceeded_precheck(int64_t consume) const { return _limit >= 0 && _limit < consumption() + consume; }
```

Comment:
Why only check single tracker not one tree?

Reply:
Done